modules/asset-manager-storage-aws/src/main/java/org/opencastproject/assetmanager/aws/s3/AwsS3AssetStore.java

```diff
@@ -93,6 +97,8 @@ public class AwsS3AssetStore extends AwsAbstractArchive implements RemoteAssetSt
   public static final int DEFAULT_CONNECTION_TIMEOUT = 10000;
   public static final int DEFAULT_MAX_RETRIES = 100;
 
+  public static final long DOWNLOAD_URL_EXPIRATION_MS = 30 * 60 * 1000; // 30 min
```

Comment:
Is this timeout an issue for Glacier, or is this irrelevant once a connection is opened?

Reply:
This timeout indicates for how long the signed url link is valid and I believe 30 minutes should be enough to download any files, but we could also make the timeout configurable... Glacier Instant Retrieval works fine, we  use it. I don't think our code currently supports Glacier Deep Archive and Flexible Retrieval.